lib/whitelist/whitelist.rb

```diff
@@ -0,0 +1,43 @@
+class Whitelist
+
+  def initialize(whitelist)
+    @whitelist = whitelist
+  end
+
+  def apply(check_name, headers, rows)
+    check = get_or_else(@whitelist, check_name, {})
+    rules = get_or_else(check, 'rules', [])
+    rule_predicates = rules.map { |rule| Whitelist.predicate_for(headers, get_or_else(rule, 'predicate', [])) }
+    rows.reject { |row| rule_predicates.any? { |r| r.call(row) } }
+  end
+
+  def get_or_else(hash, key, default)
+    # can't use #fetch since it returns nil if the key exists but maps to nil
```

Comment:
I believe the default you can provide to `fetch` is used when the key is missing, but not when it is present and maps to the value `nil`.
'Billion dollar mistake' was a large underestimate :P